src/base/rss/rss_feed.cpp

```diff
@@ -207,13 +207,14 @@ void Feed::handleParsingFinished(const RSS::Private::ParsingResult &result)
         m_lastBuildDate = result.lastBuildDate;
 
         foreach (const QVariantHash &varHash, result.articles) {
-            auto article = Article::fromVariantHash(this, varHash);
-            if (article) {
+            try {
+                auto article = new Article(this, varHash);
                 if (!addArticle(article))
                     delete article;
                 else
                     m_dirty = true;
             }
+            catch (const std::logic_error&) {}
```

Comment:
No, it is not a logic error according to your explanation. `std::logic_error' is an error that has to be corrected when debugging a program, and as such  `catch (const std::logic_error&)` is a kind of `assert()`. `std::logic_error' indicates a logical flaw in the code. Most likely you need `std::runtime_error` here (I mean you need to throw `std::runtime_error`). That exception means a problem during run-time, like invalid data on input.